pkg/api/v1alpha2/types_config.go

```diff
@@ -0,0 +1,169 @@
+	// This is currently not implemented.
+	Samples []SampleImages `json:"samples,omitempty"`
+}
+
+// OCP defines the configuration for OCP and OKD content types.
+type OCP struct {
+	// Graph defines whether Cincinnati graph data will
+	// downloaded and publish
+	Graph bool `json:"graph,omitempty"`
+	// Channels defines the configuration for individual
+	// OCP and OKD channels
+	Channels []ReleaseChannel `json:"channels,omitempty"`
+}
+
+// ReleaseChannel defines the configuration for individual
+// OCP and OKD channels
+type ReleaseChannel struct {
+	Name string `json:"name"`
+	// MinVersion is minimum version in the
+	// release channel to mirror
+	MinVersion string `json:"minVersion"`
+	// MaxVersion is maximum version in the
+	// release channel to mirror
+	MaxVersion string `json:"maxVersion"`
+	// ShortestPath mode calculates the shortest path
+	// between the min and mav version
+	ShortestPath bool `json:"shortestPath,omitempty"`
+	// Full mode set the MinVersion to the
+	// first release in the channel and the MaxVersion
+	// to the last release in the channel.
+	Full bool `json:"full,omitempty"`
+}
+
+// IsHeadsOnly determine if the mode set mirrors only channel head.
+// Setting MaxVersion will override this setting.
+func (r ReleaseChannel) IsHeadsOnly() bool {
+	return !r.Full
+}
+
+// Operator defines the configuration for operator catalog mirroring.
+type Operator struct {
+	// Mirror specific operator packages, channels, and versions, and their dependencies.
+	// If HeadsOnly is true, these objects are mirrored on top of heads of all channels.
+	// Otherwise, only these specific objects are mirrored.
+	IncludeConfig `json:",inline"`
+
+	// Catalog image to mirror. This image must be pullable and available for subsequent
+	// pulls on later mirrors.
+	// This image should be an exact image pin (registry/namespace/name@sha256:<hash>)
+	// but is not required to be.
+	Catalog string `json:"catalog"`
+	// Full defines whether all packages within the catalog
+	// or specified IncludeConfig will be mirrored or just channel heads.
+	Full bool `json:"full,omitempty"`
+	// SkipDependencies will not include dependencies
+	// of bundles included in the diff if true.
+	SkipDependencies bool `json:"skipDeps,omitempty"`
+}
+
+// IsHeadsOnly determine if the mode set mirrors only channel heads of all packages in the catalog.
+// Channels specified in DiffIncludeConfig will override this setting;
+// heads will still be included, but prior versions may also be included.
+func (o Operator) IsHeadsOnly() bool {
+	return !o.Full
+}
+
+// Helm defines the configuration for Helm chart download
+// and image mirroring
+type Helm struct {
+	// Repo is the helm repository containing the charts
+	Repos []Repo `json:"repos,omitempty"`
+	// Local is the configuration for locally stored helm charts
+	Local []Chart `json:"local,omitempty"`
+}
+
+// Repo defines the configuration for a Helm repo.
+type Repo struct {
```

Comment:
Nit: the general API guidelines (https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#naming-conventions) I'd suggest renaming this to `Repositories`

Reply:
Ack. Will incorporate into a follow on PR.